### ISSUE TITLE
[Question]: TextClassifier question

### ISSUE THREAD

**Issue by alejandrojcastaneira** [zalandoresearch/flair#3680]
### Question

Hi everyone, is there a way to perform binary classification with the TextClassifier? I have a dataset with only two classes, positive and negative, and I'm loading the corpus as a .CSV file with the following format: `"text" , "label"`, the label can be either 0 or 1. However, the label dictionary creates two separate classes, and training begins in this manner. 

After fine-tuning, the model starts predicting these two classes separately. Sometimes, a single sentence can be both, which I assume is due to the Crossentropy loss used. How can I format the training data or modify the classifier to change this behaviour? Also, would it be possible to train with BinaryCrossentropy?

**Comment by samGende** [comment]
If you want to force the Classifier to only predict one class you can init with multi_label=false:

`TextClassifier(embeddings=document_embeddings, label_type=label_type, label_dictionary = label_dictionary, multi_label = False)`

That should work if your .CSV only has two columns for the text and label. 

You may also want to double check if the multiple labels are due to each sentence having a target label, and predicted label. When predicting the class of a sentence flair adds that prediction as a label to the Sentence object. You can specify the name of the label like this: 
`classifier.predict(sentence, label_name="pred")`

**Comment by jbbqqf** [comment]
Hi! Drive-by triage of older issues — I think this can be closed; the answer that `@samGende` gave above is the right one.

For single-label / binary classification with `TextClassifier`, you force single-label mode by passing `multi_label=False` at construction time. The double-prediction you're seeing is just the default auto-detect resolving to multi-label on a tiny 2-class corpus.

Evidence:
- `TextClassifier` documents the `multi_label` argument as "auto-detected by default, but you can set this to True to force multi-label predictions or False to force single-label predictions": https://github.com/flairNLP/flair/blob/d4ea3777998ba67bfbe6b6b8359e024dcf673c3e/flair/models/text_classification_model.py#L36
- The parameter is propagated through `**classifierargs` into the base classifier (`DefaultClassifier.__init__`), so passing `multi_label=False` to `TextClassifier(...)` is the supported pattern: https://github.com/flairNLP/flair/blob/d4ea3777998ba67bfbe6b6b8359e024dcf673c3e/flair/models/text_classification_model.py#L23-L42
- For BCE loss specifically, Flair's training stack is built around cross-entropy / BCE-with-logits depending on `multi_label`; there isn't a separate "BinaryCrossentropy" knob to flip beyond that flag.

If `multi_label=False` didn't actually fix the predictions on your data, reopen with a minimal repro and I (or the maintainers) can dig further. Otherwise would you mind closing this out?

---
*Disclosure: I drafted this comment with help from Claude Code while triaging stale issues; the references above were verified manually.*
